lib/cartopy/mpl/gridliner.py

```diff
@@ -833,12 +833,17 @@ def _axes_domain(self, nx=None, ny=None):
         ax_transform = self.axes.transAxes
         desired_trans = ax_transform - transform
 
-        nx = nx or 100
-        ny = ny or 100
+        nx = nx or 101
+        ny = ny or 101
         x = np.linspace(1e-9, 1 - 1e-9, nx)
         y = np.linspace(1e-9, 1 - 1e-9, ny)
         x, y = np.meshgrid(x, y)
 
+        xyp = (self.axes.patch.get_transform()-ax_transform).transform(
+            self.axes.patch.get_path().vertices)
+        x = np.concatenate((x.ravel(), xyp[:, 0]))
+        y = np.concatenate((y.ravel(), xyp[:, 1]))
+
         coords = np.column_stack((x.ravel(), y.ravel()))
```

Comment:
The `.ravel()` should no longer be needed here after the previous step.

Reply:
`x` and `y` are 2D after meshgrid, so we to ravel them before concatenation.

Reply:
But you ravel them on 844/845 already.

Reply:
I fixed it.